downstairs/src/dynamometer.rs

```diff
@@ -0,0 +1,204 @@
+                        + (total_io_time.subsec_nanos() as f32 / 1e9);
+
+                    iops.push(io_operations_sent as f32 / fractional_seconds);
+                    bws.push(bw_consumed as f32 / fractional_seconds);
+
+                    io_operations_sent = 0;
+                    bw_consumed = 0;
+                    measurement_time = Instant::now();
+                    total_io_time = Duration::ZERO;
+
+                    if iops.len() >= samples {
+                        break 'outer;
+                    }
+                }
+
+                match flush_config {
+                    DynoFlushConfig::FlushPerIops(value) => {
+                        if iops_since_last_flush > value {
+                            region
+                                .region_flush(
+                                    flush_number,
+                                    gen_number,
+                                    &None, // snapshot_details
+                                    JobId(1000),
+                                    None, // extent_limit
+                                )
+                                .await?;
+
+                            flush_number += 1;
+                            gen_number += 1;
+                            iops_since_last_flush = 0;
+                        }
+                    }
+
+                    DynoFlushConfig::FlushPerBlocks(value) => {
+                        if blocks_since_last_flush > value {
+                            region
+                                .region_flush(
+                                    flush_number,
+                                    gen_number,
+                                    &None, // snapshot_details
+                                    JobId(1000),
+                                    None, // extent_limit
+                                )
+                                .await?;
+
+                            flush_number += 1;
+                            gen_number += 1;
+                            blocks_since_last_flush = 0;
+                        }
+                    }
+
+                    DynoFlushConfig::FlushPerMs(value) => {
+                        if flush_time.elapsed()
+                            > Duration::from_millis(value as u64)
+                        {
+                            region
+                                .region_flush(
+                                    flush_number,
+                                    gen_number,
+                                    &None, // snapshot_details
+                                    JobId(1000),
+                                    None, // extent_limit
+                                )
+                                .await?;
+
+                            flush_number += 1;
+                            gen_number += 1;
+                            flush_time = Instant::now();
+                        }
+                    }
+                }
```

Comment:
How about factoring out the actual flush, e.g.
```suggestion
                let needs_flush = match flush_config {
                    DynoFlushConfig::FlushPerIops(value)
                        if iops_since_last_flush > value =>
                    {
                        iops_since_last_flush = 0;
                        true
                    }

                    DynoFlushConfig::FlushPerBlocks(value)
                        if blocks_since_last_flush > value =>
                    {
                        blocks_since_last_flush = 0;
                        true
                    }

                    DynoFlushConfig::FlushPerMs(value)
                        if flush_time.elapsed()
                            > Duration::from_millis(value as u64) =>
                    {
                        flush_time = Instant::now();
                        true
                    }
                    _ => false,
                };
                if needs_flush {
                    region
                        .region_flush(
                            flush_number,
                            gen_number,
                            &None, // snapshot_details
                            JobId(1000),
                            None, // extent_limit
                        )
                        .await?;

                    flush_number += 1;
                    gen_number += 1;
                }
```

Reply:
this is really neat, done in 57854da